include/boost/beast/core/string.hpp

```diff
@@ -10,82 +10,17 @@
 #ifndef BOOST_BEAST_STRING_HPP
 #define BOOST_BEAST_STRING_HPP
 
-#include <boost/beast/core/detail/config.hpp>
-#include <boost/version.hpp>
-
-#if defined(BOOST_BEAST_USE_STD_STRING_VIEW)
-#include <string_view>
-#else
-#include <boost/utility/string_view.hpp>
-#endif
-
-#include <algorithm>
+#include <boost/beast/core/detail/string.hpp>
 
 namespace boost {
 namespace beast {
 
-#if defined(BOOST_BEAST_USE_STD_STRING_VIEW)
-  /// The type of string view used by the library
-  using string_view = std::string_view;
-
-  /// The type of basic string view used by the library
-  template<class CharT, class Traits>
-  using basic_string_view =
-      std::basic_string_view<CharT, Traits>;
-#else
-  /// The type of string view used by the library
-  using string_view = boost::string_view;
-
-  /// The type of basic string view used by the library
-  template<class CharT, class Traits>
-  using basic_string_view =
-      boost::basic_string_view<CharT, Traits>;
-#endif
-
-namespace detail {
-
-inline
-char
-ascii_tolower(char c)
-{
-    return ((static_cast<unsigned>(c) - 65U) < 26) ?
-        c + 'a' - 'A' : c;
-}
-
-template<class = void>
-bool
-iequals(
-    beast::string_view lhs,
-    beast::string_view rhs)
-{
-    auto n = lhs.size();
-    if(rhs.size() != n)
-        return false;
-    auto p1 = lhs.data();
-    auto p2 = rhs.data();
-    char a, b;
-    // fast loop
-    while(n--)
-    {
-        a = *p1++;
-        b = *p2++;
-        if(a != b)
-            goto slow;
-    }
-    return true;
-slow:
-    do
-    {
-        if(ascii_tolower(a) != ascii_tolower(b))
-            return false;
-        a = *p1++;
-        b = *p2++;
-    }
-    while(n--);
-    return true;
-}
+/// The type of string view used by the library
+using detail::string_view;
```

Comment:
The docs will show `detail::string_view` since that's what you put there, which is not ideal. Instead, the aliases should go into **core/string_type.hpp** instead of **core/detail/string.hpp**. I'll make this change before I merge

Reply:
wouldn't `using string_view = detail::string_view` work?